### PR TITLE
setup/ PWA structure

### PR DESCRIPTION
**Pull request by dietza**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

including setup of the *service worker and registration* files,
early updates to boilerplates & imports where necessary,
establishes basic component file architecture
> Also includes an unrelated update to the PR template

All of `src/worker.js` and `src/workerRegistration.js`
Line 27 in index.html
Line 2 in manifest.json
Line 3 in index.js

After reviewing issue discussion on GH around a breaking change that may have presented some of the problems we were experiencing, we changed course to initialize a separate repo and run the concurrent set up commands
`npx create-react-app <scrap-app-name> --template cra-template-pwa` so that we could access some of the boilerplate code and transfer that to new files in our existing app.

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/worker.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import { precacheAndRoute, createHandlerBoundToURL } from 'workbox-precaching';
+import { registerRoute } from 'workbox-routing';
+import { StaleWhileRevalidate } from 'workbox-strategies';
+
+clientsClaim();
+
+// Precache all of the assets generated by your build process.
+// Their URLs are injected into the manifest variable below.
+// This variable must be present somewhere in your service worker file,
+// even if you decide not to use precaching. See https://cra.link/PWA
```
Comment (colefiscus) [inline src/worker.js:21]: Let's delete some unnecessary boilerplate comments before we merge?

File: src/worker.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/* eslint-disable no-restricted-globals */
```
Comment (colefiscus) [inline src/worker.js:1]: I don't think this matters, but I think the file is called service-worker.js in the template boilerplate. Something to remember, maybe?

File: src/worker.js (hunk 1 above)
Comment (dietza) [reply src/worker.js:21]: for sure! May leave some that could be helpful as annotations for the time being, but will clear these instructive comments.

File: src/worker.js (hunk 2 above)
Comment (dietza) [reply src/worker.js:1]: 👍

**Pull request merged**